hathor/consensus.py

```diff
@@ -249,17 +249,40 @@ def update_voided_info(self, block: Block) -> None:
                 else:
                     storage._best_block_tips = [blk.hash for blk in heads]
                     # XXX Is it safe to select one of the heads?
-                    best_block = heads[0]
-                    assert best_block.hash is not None
-                    best_meta = best_block.get_metadata()
-                    self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
-                    storage.add_new_to_block_height_index(best_meta.height, best_block.hash)
+                    # best_block = heads[0]
+                    # assert best_block.hash is not None
+                    # best_meta = best_block.get_metadata()
+                    # self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
+                    # storage.add_new_to_block_height_index(best_meta.height, best_block.hash, block.timestamp)
```

Comment:
I'm bringing this up from https://github.com/HathorNetwork/hathor-core/pull/226#discussion_r654156126 (which was moved to #238

Commenting this has no impact on any tests except possible simulator tests, but I haven't singled out a seed that fails with this commented but passes if it isn't (or the other way around).

So at the very least we still need to test this properly.

Reply:
As far as I remember, I tested all possible scenarios here. Can you comment the line `storage._best_block_tips = [blk.hash for blk in heads]` and try again? It should break at least one test.

Reply:
I'm not sure what's the best behavior here. We currently void all blocks in this situation.

Reply:
Fixed in #292 .